src/ctapipe/calib/camera/tests/test_extractor.py

```diff
@@ -0,0 +1,136 @@
+"""
+Tests for StatisticsExtractor and related functions
+"""
+
+import numpy as np
+import pytest
+from astropy.table import Table
+from astropy.time import Time
+
+from ctapipe.calib.camera.extractor import PlainExtractor, SigmaClippingExtractor
+
+
+def test_extractors(example_subarray):
+    """test basic functionality of the StatisticsExtractors"""
+
+    # Create dummy data for testing
+    times = Time(
+        np.linspace(60117.911, 60117.9258, num=5000), scale="tai", format="mjd"
+    )
+    event_ids = np.linspace(35, 725000, num=5000, dtype=int)
+    ped_data = np.random.normal(2.0, 5.0, size=(5000, 2, 1855))
+    charge_data = np.random.normal(77.0, 10.0, size=(5000, 2, 1855))
+    time_data = np.random.normal(18.0, 5.0, size=(5000, 2, 1855))
+    # Create tables
+    ped_table = Table(
+        [times, event_ids, ped_data],
+        names=("time_mono", "event_id", "image"),
+    )
+    charge_table = Table(
+        [times, event_ids, charge_data],
+        names=("time_mono", "event_id", "image"),
+    )
+    time_table = Table(
+        [times, event_ids, time_data],
+        names=("time_mono", "event_id", "peak_time"),
+    )
+    # Initialize the extractors
+    chunk_size = 2500
+    ped_extractor = SigmaClippingExtractor(
+        subarray=example_subarray, chunk_size=chunk_size
+    )
+    ff_charge_extractor = SigmaClippingExtractor(
+        subarray=example_subarray, chunk_size=chunk_size
+    )
+    ff_time_extractor = PlainExtractor(subarray=example_subarray, chunk_size=chunk_size)
+
+    # Extract the statistical values
+    ped_stats = ped_extractor(table=ped_table)
+    charge_stats = ff_charge_extractor(table=charge_table)
+    time_stats = ff_time_extractor(table=time_table, col_name="peak_time")
+
+    # Check if the start and end values are properly set for the timestamps and event IDs
+    assert ped_stats[0]["time_start"] == times[0]
+    assert time_stats[0]["event_id_start"] == event_ids[0]
+    assert ped_stats[1]["time_end"] == times[-1]
+    assert time_stats[1]["event_id_end"] == event_ids[-1]
+
+    # Check if the calculated statistical values are reasonable
+    # for a camera with two gain channels
+    assert not np.any(np.abs(ped_stats[0]["mean"] - 2.0) > 1.5)
```

Comment:
These tests are not very helpful if they fail.

Using the numpy test functions yields much better error messages, e.g. in this case, I would test:

```
np.testing.assert_allclose(ped_stats[0]["mean"], 2.0, atol=..., rtol=...)
```